### DISCUSSION TITLE
What's the easiest way to automate adding transactions? Email scrape? Some kind of API connection with the bank? Something else?

### DISCUSSION THREAD

**Discussion by AkshayRao27** [firefly-iii/firefly-iii#5431] [Firefly III]
I set up a Firefly instance in Docker on my Linux server recently and was wondering if there's any way to automatically create transactions in Firefly whenever I make a payment. 

When I make a payment or transfer, the bank sends me an email with the transfer details. Is there a way to perhaps forward this email to Firefly, have the date scraped, and automatically added as a transaction? 

Or can this somehow be done via an API? I'm with Scotiabank in Canada and Saltedge says its a supported bank. 

Is this sort of functionality even possible on a regular basis or is the data import functionality intended for just a one time bulk import and not for a per-transaction basis? 

I am very new to all of this hosting stuff and have little to no development knowledge, but I'm willing to learn. However, if this involves very heavy programming or coding, please keep in mind that I'm going in as a complete noob :)

**Comment by JC5** [comment]
You already found Salt Edge, so that's a positive thing.

The Data Importer can and should run regularly, to keep up with new transactions. I run mine daily. 

Just like the Docker instance you set up for Firefly III, you can set up the Data Importer in Docker and access it through your browser.

**Reply by LM3469** [reply]
Hello @JC5 
Does anyone have created a tutorial to detail how to automate the data importer task daily?
I have FIDI on Docker using a Synology NAS.
I am more in the "noob" part of us all and willing to learn but I haven't find a way yet.
Thanks!

**Reply by JC5** [reply]
Save your JSON somewhere, and use the command line in a cron job: https://docs.firefly-iii.org/data-importer/usage/command_line/
